pallets/gear/src/tests.rs

```diff
@@ -14865,6 +14867,68 @@ fn incorrect_store_context() {
     });
 }
 
+#[test]
+fn allocate_in_init_free_in_handle() {
+    let wat = r#"
+        (module
+            (import "env" "memory" (memory 16))
+            (import "env" "alloc" (func $alloc (param i32) (result i32)))
+            (import "env" "free" (func $free (param i32) (result i32)))
+            (export "init" (func $init))
+            (export "handle" (func $handle))
+            (func $init
+                (call $alloc (i32.const 1))
+                drop
+            )
+            (func $handle
+                (call $free (i32.const 16))
+                drop
+            )
+        )
+    "#;
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        assert_ok!(Gear::upload_program(
+            RuntimeOrigin::signed(USER_1),
+            ProgramCodeKind::Custom(wat).to_bytes(),
+            DEFAULT_SALT.to_vec(),
+            vec![],
+            1_000_000_000,
+            0,
+            false,
+        ));
+
+        let program_id = get_last_program_id();
+
+        run_to_next_block(None);
+
+        let Some(Program::Active(program)) = ProgramStorageOf::<Test>::get_program(program_id)
+        else {
+            panic!("program must be active")
+        };
+        assert_eq!(program.allocations, BTreeSet::from([WasmPage::from(16)]));
```

Comment:
would be nice if you clarify at least in comments or by defining a separate variable like `let pages = 16;` which will be inserted into wat, so readers understand immediately why exactly these constants are asserted.

Reply:
Sure, done